include/nlohmann/detail/macro_scope.hpp

```diff
@@ -242,6 +242,63 @@
         e = ((it != std::end(m)) ? it : std::begin(m))->first;                                  \
     }
 
+NLOHMANN_JSON_NAMESPACE_BEGIN
+namespace detail
+{
+template<typename T>
+[[noreturn]] inline void json_throw_from_serialize_macro(T&& exception)
+{
+#if defined(__cpp_exceptions) || defined(__EXCEPTIONS) || defined(_CPPUNWIND) || defined(EXCEPTIONS)
```

Comment:
When defining `JSON_THROW` in the same file, we use the following code to detect exceptions. Please use the same here to avoid issues.

```cpp
#if (defined(__cpp_exceptions) || defined(__EXCEPTIONS) || defined(_CPPUNWIND)) && !defined(JSON_NOEXCEPTION)
```

Reply:
oh okay, thanks!